seastar/src/timer.cc

```diff
@@ -3,6 +3,58 @@
 namespace seastar_ffi {
 namespace timer {
 
+// This class was created to drop callbacks passed to timer::set_calback at the right
+// time. We pass an instance of callback_object instead of a pure callback and the
+// callback is dropped when the callback_object's destrucotr is called.
+class callback_object {
+    uint8_t* _callback;
+    rust::Fn<void(uint8_t*)> _caller;
+    rust::Fn<void(uint8_t*)> _dropper;
+    bool _moved = false;
```

Comment:
Nit: I'm not sure that `_moved` is the right name. For example, the default-constructed callback_object sets it to `true`, but it wasn't really moved from. How about `_valid` (+ flipping its value)?

Reply:
Yes, `_valid` is much better.